xarray/core/options.py

```diff
@@ -26,6 +26,10 @@ class set_options(object):
     """
     def __init__(self, **kwargs):
         self.old = OPTIONS.copy()
+        for key in kwargs:
```

Comment:
please rebase or merge master -- I actually already added in something very similar
https://github.com/pydata/xarray/blob/master/xarray/core/options.py